charts/replicated-library/templates/lib/_collector_templates.tpl

```diff
@@ -0,0 +1,221 @@
+    {{ if .ContextValues.collector.selector -}}
+    selector:
+      {{- range $k, $v := .ContextValues.collector.selector }}
+        {{- if and (eq $k "app") (index $.Values.apps $v) }}
+          {{- $_ := set $.ContextNames "app" $v -}}
+          {{ include "replicated-library.labels.selectorLabels" $ | nindent 4 }}
+          {{- $_ := unset $.ContextNames "app"  -}}
+        {{- else -}}
+        {{ $k | nindent 6  }}: {{ $v }}
+        {{- end }}
+      {{- end }}
+    {{- end }}
+    {{- if .ContextValues.collector.namespace }}
+    namespace: {{ .ContextValues.collector.namespace }}
+    {{- end }}
+    {{- if .ContextValues.collector.containerNames }}
+    containerNames:
+        {{- range .ContextValues.collector.containerNames }}
+      - {{ . }}
+        {{- end }}
+    {{- end }}
+    {{ if .ContextValues.collector.limits }}
+    limits:
+      maxAge: {{ default "720h" .ContextValues.collector.limits.maxAge }}
+      maxLines: {{ default 10000 .ContextValues.collector.limits.maxLines }}
+      maxBytes: {{ default 5000000 .ContextValues.collector.limits.maxBytes }}
+    {{- else -}}
+    limits:
+      maxAge: "720h"
+      maxLines: 10000
+      maxBytes: 5000000
+    {{- end }}
+{{- end }}
+
+{{- define "replicated-library.troubleshoot.collector.exec" -}}
+- {{ .ContextNames.collector }}:
+    collectorName: {{ default "" .ContextValues.collector.collectorName }}
+    name: {{ default "" .ContextValues.collector.name }}
+    containerName: {{ default "" .ContextValues.collector.containerName }}
+    timeout: {{ default "10s" .ContextValues.collector.timeout }}
+
+    {{- if .ContextValues.collector.args }}
+    args:
+      {{- range .ContextValues.collector.args }}
+      - {{ . }}
+      {{- end }}
+    {{- end }}
+
+    {{- if .ContextValues.collector.command }}
+    command:
+      {{- range .ContextValues.collector.command }}
+      - {{ . }}
+      {{- end }}
+    {{- end }}
+
+    {{- if .ContextValues.collector.selector }}
+    selector:
+      {{- range $k, $v := .ContextValues.collector.selector }}
+        {{- if and (eq $k "app") (index $.Values.apps $v) }}
+          {{- $_ := set $.ContextNames "app" $v -}}
+          {{ include "replicated-library.labels.selectorLabels" $ | nindent 6 }}
+          {{- $_ := unset $.ContextNames "app"  -}}
+        {{- else -}}
+        {{ $k | nindent 6  }}: {{ $v }}
+        {{- end }}
+      {{- end }}
+    {{- end }}
+
+    {{- if .ContextValues.collector.namespace }}
+    namespace: {{ .ContextValues.collector.namespace }}
+    {{- end }}
+
+    {{- if .ContextValues.collector.exclude }}
+    exclude: {{ .ContextValues.collector.exclude }}
+    {{- end }}
+
+{{- end }}
+
+
+{{- define "replicated-library.troubleshoot.collector.secret" -}}
+- {{ .ContextNames.collector }}:
+    collectorName: {{ default "" .ContextValues.collector.collectorName }}
+    name: {{ default "" .ContextValues.collector.name }}
```

Comment:
This is a bit tricky, but this is the name of a secret and we need to check to see if we should add a prefix to it. The logic for that should be: If this secret is defined and enabled in this chart we add the prefix from the names template to the user input.

You can see this done on ingress with the `serviceName` field as one example that David has just recently implemented. There are other examples, I believe Statefulsets do this for the `serviceName` field as well.